packages/wallets/src/features/cashier/flows/WalletWithdrawal/WalletWithdrawal.tsx

```diff
@@ -1,27 +1,48 @@
 import React, { useEffect, useState } from 'react';
-import { WithdrawalFiatModule, WithdrawalVerificationModule } from '../../modules';
+import { useAuthorize, useCurrencyConfig } from '@deriv/api';
+import { WithdrawalCryptoModule, WithdrawalFiatModule, WithdrawalVerificationModule } from '../../modules';
+import { Loader } from '../../../../components/Loader';
 
 const WalletWithdrawal = () => {
+    const {
+        data: authorizeData,
+        isLoading: isAuthorizeLoading,
+        isSuccess: isAuthorizeSuccess,
+        switchAccount,
+    } = useAuthorize();
+    const { getConfig, isLoading: isCurrencyConfigLoading, isSuccess: isCurrencyConfigSuccess } = useCurrencyConfig();
     const [verificationCode, setVerificationCode] = useState('');
 
     useEffect(() => {
         const queryParams = new URLSearchParams(location.search);
         const verificationQueryParam = queryParams.get('verification');
+        const walletIdParam = queryParams.get('walletid');
 
-        if (verificationQueryParam) {
+        if (verificationQueryParam && walletIdParam) {
             setVerificationCode(verificationQueryParam);
+            switchAccount(walletIdParam); // authorize the intended wallet
```

Comment:
Here, we are getting the loginid from the `walletid` param and authorizing the wallet before rendering the intended fiat/crypto withdrawal module. Then we are clearing the param.

Reply:
CMIIW, so if we switch account in wallet (different connection), but we are getting the loginid from param which comes from client-store (different connection). So we will switch to that account instead of the current wallet account?

Reply:
Hi @farhan-nurzi-deriv, we extract the loginid from params and then store it in the localStorage. And I found out that upon redirection (from redirect.jsx) we are authorizing the account using the loginid in localStorage with the help of useAuthorize hook. So I think we don't need the above changes in redirect.jsx 😅. I reset these changes and instead use the loginid from localStorage. Thanks for pointing out 🙏.